container/src/main/java/com/flipkart/poseidon/Poseidon.java

```diff
@@ -82,12 +83,14 @@
     private final Application application;
     private final ExecutorService dataSourceES;
     private final ExecutorService filterES;
+    private final ApplicationContext context;
     private Server server;
 
     @Autowired
-    public Poseidon(Configuration configuration, Application application) {
+    public Poseidon(Configuration configuration, Application application, ApplicationContext context) {
```

Comment:
In sample, we wire Poseidon dependencies through web.xml (spring beans) where we explicitly wire configuration, application. May be that needs to change to pass context. Can you check by running [poseidon sample](https://github.com/flipkart-incubator/Poseidon/tree/master/sample#poseidon-sample)

Reply:
made the class ApplicationContextAware